src/cplscheme/CouplingData.hpp

```diff
@@ -91,6 +91,41 @@ class CouplingData {
   /// move to next window and initialize data via extrapolation
   void moveToNextWindow();
 
+  /**
+   * @brief Returns the values of all time steps stored in this coupling data in a serialized fashion
+   *
+   * Serialization of the data is performed per mesh node. The dimension of one node is then data dimension (scalar or vector) * number of time steps.
+   *
+   * @return Eigen::VectorXd a vector containing all data for all time steps in serialized fashion.
+   */
+  Eigen::VectorXd getSerializedValues();
+
+  /**
+   * @brief Returns the gradients of all time steps stored in this coupling data in a serialized fashion
+   *
+   * Serialization of the data is performed per mesh node. The dimension of one node is then data dimension (scalar or vector) * number of time steps.
+   *
+   * @return Eigen::VectorXd a vector containing all gradient data for all time steps in serialized fashion.
+   */
+  Eigen::VectorXd getSerializedGradients();
+
+  /**
+   * @brief accepts serialized values and stores them in this coupling data
+   *
+   * @param timesAscending times associated with data
+   * @param serializedValues values in serialized form
+   */
+  void storeFromSerialized(Eigen::VectorXd timesAscending, Eigen::VectorXd serializedValues);
+
+  /**
+   * @brief accepts serialized values and gradients and stores them in this coupling data
+   *
+   * @param timesAscending times associated with data
+   * @param serializedValues values in serialized form
+   * @param serializedGradients gradients in serialized form
+   */
+  void storeFromSerialized(Eigen::VectorXd timesAscending, Eigen::VectorXd serializedValues, Eigen::MatrixXd serializedGradients);
```

Comment:
It's a bit odd that the "store" methods are not symmetric to the "get" methods.

Instead of `getSerializedX()`, there could also be one `serialize()`.

Compare:

https://github.com/precice/precice/blob/d3485cd614f29f82e54269028be8a737f9325954/src/com/SerializedMesh.hpp#L21

Reply:
I think the best solution is to use the same pattern as `SerializedMesh` for `SerializedStamples` (and add this class in `com`) and then do something like the following inside `BaseCouplingScheme::sendData`

```
const auto serializedStamples = com::SerializedStamples::serialize(data->stamples());

// Data is actually only send if size>0, which is checked in the derived classes implementation
m2n->send(serializedStamples.values(), data->getMeshID(), data->getDimensions() * serializedStamples.nTimeSteps);

if (data->hasGradient()) {
  m2n->send(serializedStamples.gradients(), data->getMeshID(), data->getDimensions() * data->meshDimensions() * serializedStamples.nTimeSteps());
}
```

and `BaseCouplingScheme::receiveData`:
 
```
int             nTimeSteps = 2;
Eigen::VectorXd timesAscending(nTimeSteps);
timesAscending << time::Storage::WINDOW_START, time::Storage::WINDOW_END;
com::SerializedStamples = serializedStamples();

// Data is only received on ranks with size>0, which is checked in the derived class implementation
serializedStamples.allocateValues(nTimeSteps * data->getSize());
m2n->receive(serializedStamples.values(), data->getMeshID(), data->getDimensions() * nTimeSteps);

if (data->hasGradient()) {
  serializedStamples.allocateGradients(nTimeSteps * data->getSize() * data->meshDimensions());
  m2n->receive(serializedStamples.gradients(), data->getMeshID(), data->getDimensions() * data->meshDimensions() * nTimeSteps);
}

data->stamples() = serializedStamples.toStamples();
```

@uekerman does this look reasonable?

Reply:
Implementation of this part is done.